builder/buffer.go

```diff
@@ -134,17 +147,11 @@ func (b Buffer) escape(table, value string) string {
 	var escaped_table string
 	if table != "" {
 		if i := strings.Index(strings.ToLower(table), " as "); i > -1 {
-			return b.escape(table[:i], "") + " AS " + b.Quoter.ID(table[i+4:])
-		}
-		if b.AllowTableSchema && strings.IndexByte(table, '.') >= 0 {
-			parts := strings.Split(table, ".")
-			for i, part := range parts {
-				part = strings.TrimSpace(part)
-				parts[i] = b.Quoter.ID(part)
-			}
-			escaped_table = strings.Join(parts, ".")
+			escaped_table = b.escape_schema(table[:i]) + " AS " + b.Quoter.ID(strings.TrimSpace(table[i+4:]))
+		} else if i := strings.Index(strings.ToLower(table), " "); i > -1 {
```

Comment:
this will break if the table name contains spaces

Reply:
I'm not saying we should force user to use AS, but we need to find a good way if we want to support that


As for now, I think bug in example 2 is more critical